File: GPSurb/GPSurb/Model/ResultsModel.swift
//
//  ResultsModel.swift
//  GPSurb
//
//  Created by Gilson Santos on 16/10/19.
//  Copyright © 2019 Gilson Santos. All rights reserved.
//

import Foundation

struct ResultsModel: Codable {
    let sku: String?
    let isHotel: Bool?
    let category: String?
    let smallDescription: String?
    let amenities: [AmenitiesModel]?
    let idResult: String?
    let price: PriceModel?
    let huFreeCancellation: Bool?
    let image: String?
    let name: String?
    let url: String?
    let description: String?
    let stars: Int?
    let gallery: [GalleryModel]?
    let address: AddressModel?
    let tags: [String]?
    let quantityDescriptors: QuantityDescriptorsModel?
    let featuredItem: FeaturedItemModel?
    
    enum CodingKeys: String, CodingKey {
        case sku = "sku"
        case isHotel = "isHotel"
        case category = "category"
        case smallDescription = "smallDescription"
        case amenities = "amenities"
        case idResult = "id"
        case price = "price"
        case huFreeCancellation = "hu_free_cancellation"
        case image = "image"
        case name = "name"
        case url = "url"
        case description = "description"
        case stars = "stars"
        case gallery = "gallery"
        case address = "address"
        case tags = "tags"
        case quantityDescriptors = "quantityDescriptors"
        case featuredItem = "featuredItem"
    }
}

File: app/src/main/java/viniciusapp/com/br/hurbtest/models/ResultsModel.kt
package viniciusapp.com.br.hurbtest.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class ResultsModel(
    @field:SerializedName("sku")
    var sku: String? = null,
    @field:SerializedName("isHotel")
    var isHotel: Boolean? = true,
    @field:SerializedName("category")
    var category: String? = null,
    @field:SerializedName("smallDescription")
    var smallDescription: String? = null,
    @field:SerializedName("amenities")
    var amenities: List<AmenitiesModel>? = null,
    @field:SerializedName("id")
    var id: String? = null,
    @field:SerializedName("price")
    var price: PriceModel? = null,
    @field:SerializedName("hu_free_cancellation")
    var hu_free_cancellation: Boolean? = null,
    @field:SerializedName("image")
    var image: String? = null,
    @field:SerializedName("name")
    var nameHotel: String? = null,
    @field:SerializedName("url")
    var url: String? = null,
    @field:SerializedName("description")
    var description: String? = null,
    @field:SerializedName("stars")
    var stars: String? = null,
    @field:SerializedName("gallery")
    var gallery: ArrayList<GalleryModel>? = null,
    @field:SerializedName("address")
    var address: AddressModel? = null,
    @field:SerializedName("tags")
    var tags: List<String>? = null,
    @field:SerializedName("quantityDescriptors")
    var quantityDescriptors: QuantityDescriptorsModel? = null,
    @field:SerializedName("featuredItem")
    var featuredItem: FeaturedItemModel? = null
): Serializable